engine/src/engine.rs

```diff
@@ -958,7 +965,7 @@ pub fn refund_on_error<I: IO + Copy, E: Env, P: PromiseHandler>(
     if let Some(erc20_address) = args.erc20_address {
         // ERC-20 exit; re-mint burned tokens
         let erc20_admin_address = current_address(&current_account_id);
-        let mut engine =
+        let mut engine: Engine<I, E, aurora_engine_modexp::AuroraModExp> =
```

Comment:
Above the engine is instantiated with a generic `ModexpAlgorithm` type parameter (e.g. in `submit`), but here a concrete implementation is used. Why?

Reply:
Because another implementation of `ModExp` is needed in the Borealis infrastructure only.